src/garage/tf/algos/rl2.py

```diff
@@ -7,89 +7,167 @@
 
 import akro
 from dowel import logger
-import gym
 import numpy as np
 
-from garage import log_multitask_performance, TrajectoryBatch
-from garage.envs import EnvSpec
+from garage import (Environment, EnvSpec, EnvStep, log_multitask_performance,
+                    StepType, TrajectoryBatch)
 from garage.misc import tensor_utils as np_tensor_utils
 from garage.np.algos import MetaRLAlgorithm
 from garage.sampler import DefaultWorker
 from garage.tf.algos._rl2npo import RL2NPO
 
 
-class RL2Env(gym.Wrapper):
+class RL2Env(Environment):
```

Comment:
you may want to introduce your own EnvironmentWrapper type to make this easier (it's okay to leave an issue instead)